src/tools/kata-ctl/src/ops/env_ops.rs

```diff
@@ -0,0 +1,445 @@
+
+    let support_v_socks = utils::supports_vsocks(utils::VHOST_VSOCK_DEVICE)?;
+
+    Ok(HostInfo {
+        kernel: host_kernel_version,
+        architecture: String::from(std::env::consts::ARCH),
+        distro: host_distro,
+        cpu: host_cpu,
+        memory: memory_info,
+        available_guest_protection: guest_protection,
+        vm_container_capable: true, // TODO: implement
+        support_v_socks,
+    })
+}
+
+pub fn get_runtime_info(toml_config: &TomlConfig) -> RuntimeInfo {
+    let version = VersionInfo {
+        semver: String::from(version::VERSION),
+        commit: String::from(version::COMMIT),
+    };
+
+    let config_path = TomlConfig::get_default_config_file();
+    let mut toml_path = String::new();
+    if config_path.is_ok() {
+        let p = config_path.unwrap();
+        let path_str = p.to_str();
+        toml_path = match path_str {
+            Some(s) => String::from(s),
+            None => String::new(),
+        };
+    }
+
+    RuntimeInfo {
+        // TODO: Needs to be implemented
+        path: "TBD".to_string(),
+        version,
+        experimental: toml_config.runtime.experimental.clone(),
+        guest_se_linux_label: String::from("not implemented yet"),
+        debug: toml_config.runtime.debug,
+        trace: toml_config.runtime.enable_tracing,
+        disable_guest_seccomp: toml_config.runtime.disable_guest_seccomp,
+        disable_new_net_ns: toml_config.runtime.disable_new_netns,
+        sandbox_cgroup_only: toml_config.runtime.sandbox_cgroup_only,
+        static_sandbox_resource_mgmt: toml_config.runtime.static_sandbox_resource_mgmt,
+        config: RuntimeConfigInfo { path: toml_path },
+    }
+}
+
+pub fn get_agent_info(toml_config: &TomlConfig) -> Result<AgentInfo> {
+    let agent_config = toml_config
+        .agent
+        .get(&toml_config.runtime.agent_name)
+        .ok_or("could not find agent config in configuration")
+        .map_err(|e| anyhow!(e))?;
+
+    Ok(AgentInfo {
+        debug: agent_config.debug,
+        trace: agent_config.enable_tracing,
+    })
+}
+
+use std::process::Command;
+
+pub fn get_command_version(cmd: &str) -> Result<String> {
+    // Path is empty in case of dragonball hypervisor
+    if cmd.is_empty() {
+        return Ok("unknown".to_string());
+    }
+    let output = Command::new(cmd)
+        .arg("--version")
+        .output()
+        .map_err(|e| anyhow!(e))?;
+
+    let version = String::from_utf8(output.stdout).map_err(|e| anyhow!(e))?;
+
+    Ok(version)
+}
+
+pub fn get_hypervisor_info(
+    toml_config: &TomlConfig,
+) -> Result<(HypervisorInfo, ImageInfo, KernelInfo, InitrdInfo)> {
+    let hypervisor_config = toml_config
+        .hypervisor
+        .get(&toml_config.runtime.hypervisor_name)
+        .ok_or("could not find hypervisor config in configuration")
+        .map_err(|e| anyhow!(e))?;
+
+    let version =
+        get_command_version(&hypervisor_config.path).context("error getting hypervisor version")?;
```

Comment:
This is being displayed on multiple lines. On my systems:

## CH

  ```bash
  $ /usr/local/bin/cloud-hypervisor --version|tee /dev/tty|wc -l
  cloud-hypervisor v30.0-6-g04ee1efa-dirty
  1
  ```

But the output shown is:

```toml
[hypervisor]
machine_type = ''
machine_accelerators = ''
version = '''
cloud-hypervisor v30.0-6-g04ee1efa-dirty
'''
path = '/usr/local/bin/cloud-hypervisor'
```

## QEMU

```bash
$ /opt/kata/bin/qemu-system-x86_64 --version|tee /dev/tty|wc -l
QEMU emulator version 7.2.0 (kata-static)
Copyright (c) 2003-2022 Fabrice Bellard and the QEMU Project developers
2
```

```toml
[hypervisor]
machine_type = 'q35'
machine_accelerators = ''
version = '''
QEMU emulator version 7.2.0 (kata-static)
Copyright (c) 2003-2022 Fabrice Bellard and the QEMU Project developers
'''
path = '/opt/kata/bin/qemu-system-x86_64'
```

`kata-runtime env` displays the NL as `\n` whilst maintaining a single line:

```toml
[Hypervisor]
  MachineType = "q35"
  Version = "QEMU emulator version 7.2.0 (kata-static)\nCopyright (c) 2003-2022 Fabrice Bellard and the QEMU Project developers"
  Path = "/opt/kata/bin/qemu-system-x86_64"
```

Reply:
This shows correctly in the json output. Seems to be an issue with the serde toml encoder. Let me see if I can fix this.

Reply:
@jodh-intel This is fixed now. It was caused by `toml:to_string_pretty`. I am using toml::to_string instead now, the string now shows correctly.  I feel this is a bug with the pretty output. 
In case of toml, I see that I didnt really need the pretty output, but I do need that in case of json.